rules/no-single-promise-in-promise-methods.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+const isPromiseMethodWithArray = require('./utils/is-promise-method-with-array.js');
+
+const MESSAGE_ID_ERROR = 'no-single-promise-in-promise-methods/error';
+const MESSAGE_ID_SUGGESTION_1 = 'no-single-promise-in-promise-methods/suggestion-1';
+const MESSAGE_ID_SUGGESTION_2 = 'no-single-promise-in-promise-methods/suggestion-2';
+const messages = {
+	[MESSAGE_ID_ERROR]: 'Parameter in `Promise.{{method}}` should not be a single element array.',
+	[MESSAGE_ID_SUGGESTION_1]: 'Use the value directly.',
+	[MESSAGE_ID_SUGGESTION_2]: 'Wrap the value in a `Promise.resolve`.',
+};
+const METHODS = ['all', 'any', 'race'];
+
+const isPromiseMethodWithSinglePromise = (node, methods) => {
+	const types = new Set(['CallExpression', 'Identifier', 'MemberExpression']);
```

Comment:
We should not limit types, it can be anything.

Reply:
I authorized all types and managed the special cases of SpreadElement and SequenceExpression.